src/lib/protocols/http.c

```diff
@@ -291,17 +291,9 @@ static ndpi_protocol_category_t ndpi_http_check_content(struct ndpi_detection_mo
       if(packet->content_disposition_line.len > attachment_len) {
 	u_int8_t filename_len = packet->content_disposition_line.len - attachment_len;
 	int i;
-  if(packet->content_disposition_line.ptr[attachment_len] == '\"'){
-    //case: filename="file_name"
-    flow->http.filename = (char*)malloc((filename_len-3)*sizeof(char));
-    flow->http.filename = strncpy(flow->http.filename, (char*)packet->content_disposition_line.ptr+attachment_len+1, filename_len-2);
-  }
-  else{
-    //case: filename=file_name
-    flow->http.filename = (char*)malloc((filename_len-1)*sizeof(char));
-    flow->http.filename = strncpy(flow->http.filename, (char*)packet->content_disposition_line.ptr+attachment_len, filename_len);
-  }
 
+  flow->http.filename = ndpi_malloc(filename_len-1);
+  flow->http.filename = strncpy(flow->http.filename, (char*)packet->content_disposition_line.ptr+attachment_len, filename_len);
```

Comment:
Sorry if I wasn't clear... it seems right to strip the initial and the final `"` from the name, but you need to check both of them and remove the characters only if really present